### PR TITLE
fix: ensure header value is a string

### PR DESCRIPTION
**Pull request by ronag**
I don't think it's something else than a string in any of the branches but the check should still be correct.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by ronag** [comment]
So this was not as easy as I thought. We allow other header values than strings. I propose:

- If object, throw
- If null, empty string
- Otherwise stringify with `${val}`

**Comment by ronag** [comment]
@mcollina PTAL

File: test/client-dispatch.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ test('basic dispatch get', (t) => {
     t.equal(`localhost:${server.address().port}`, req.headers.host)
     t.equal(undefined, req.headers.foo)
     t.equal('bar', req.headers.bar)
-    t.equal('null', req.headers.baz)
+    t.equal('', req.headers.baz)
```
Comment (ronag) [inline test/client-dispatch.js:97]: @mcollina do we consider this breaking?

Comment (mcollina) [reply test/client-dispatch.js:97]: It's actually better.

**Review by KhafraDev — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**